build_tooling/prep_cpp_build.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+pushd $(realpath $(dirname $BASH_SOURCE))/../cpp/vcpkg
+
+if [[ -e "$VCPKG_INSTALLATION_ROOT" ]] ; then
+    git fetch --unshallow file://$VCPKG_INSTALLATION_ROOT
+elif [[ -n "$VCPKG_INSTALLATION_ROOT" && -e "/host$VCPKG_INSTALLATION_ROOT" ]] ; then
+    git fetch --unshallow file:///host$VCPKG_INSTALLATION_ROOT
+else
+    git fetch --unshallow origin master
+fi
+
+cd ..
+
+function windows_stuff() {
+    if [[ -n "$GITHUB_ACTION" ]] ; then
+        # Redirect the build directory to the more spacious C:
+        if [[ "$MSYSTEM" != MINGW* ]] ; then echo "Must run $0 with git/MINGW bash" >&2
+        elif [[ -e out ]] ; then echo "out directory cannot exist at this point" >&2
+        else
+            mkdir "${ARCTICDB_BUILD_DIR:?environment variable is not set}"
+            MSYS=winsymlinks:nativestrict ln -s "$ARCTICDB_BUILD_DIR" out
+        fi
+    fi
+
+    mkdir vcpkg/buildtrees vcpkg/packages out || true
+    MSYS_NO_PATHCONV=1 compact.exe /C vcpkg\\buildtrees vcpkg\\packages out
+}
+
+case `uname -a` in
+*Microsoft*)
+    # Are we in WSL to compile for Linux or is WSL bash merely used to run this script (parent process is `init`)?
```

Comment:
Can't we just assume we never use WSL to compile for Linux? Why would we ever use WSL to compile for Linux _within_ Github actions?

Reply:
I'll simplify